### PR TITLE
fix(batch): Report data error in distributed mode correctly.

### LINKED ISSUE #8066: bug: distributed scheduler execution failed, but task shows success
### Describe the bug

```

2022-03-30T09:36:48.505615Z ERROR node{id=11 name="compute-3"}:task{id=190526}: risingwave_batch::task::task_execution: Execution failed [TaskId { task_id: 5, stage_id: 1, query_id: "60c6f41d-731f-4490-9122-91a02b4b35da" }]: Expr error: Numeric out of range
2022-03-30T09:36:48.511470Z ERROR node{id=7 name="frontend-1"}:task{id=190488}: risingwave_frontend::scheduler::distributed::stage: Stage QueryId { id: "60c6f41d-731f-4490-9122-91a02b4b35da" }-1 failed to schedule tasks, error: TaskExecutionError("Expr error: Numeric out of range")
2022-03-30T09:36:48.511470Z ERROR node{id=7 name="frontend-1"}:task{id=190487}: risingwave_frontend::scheduler::distributed::query: Query stage QueryId { id: "60c6f41d-731f-4490-9122-91a02b4b35da" }-1 failed: TaskExecutionError("Expr error: Numeric out of range").
thread '<unnamed>' panicked at 'statement is expected to fail, but actually succeed:
[SQL] SELECT i.f1, i.f1 * smallint '2' AS x FROM INT2_TBL i;
at e2e_test/batch/issue_7324.slt:27
', /risingwave/src/tests/simulation/src/slt.rs:114:31
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:64:14
   2: core::panicking::panic_display
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:147:5
   3: risingwave_simulation::slt::run_slt_task::{{closure}}
             at ./src/tests/simulation/src/slt.rs:114:31
   4: risingwave_simulation::main::{{closure}}::{{closure}}::{{closure}}
             at ./src/tests/simulation/src/main.rs:188:57
   5: async_task::raw::RawTask<F,T,S>::run
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.3.0/src/raw.rs:511:20
   6: madsim::sim::task::Executor::run_all_ready
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.17/src/sim/task.rs:236:17
   7: madsim::sim::task::Executor::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.17/src/sim/task.rs:187:13
   8: madsim::sim::runtime::Runtime::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.17/src/sim/runtime/mod.rs:124:9
   9: madsim::sim::runtime::builder::Builder::run::{{closure}}::{{closure}}::{{closure}}
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.17/src/sim/runtime/builder.rs:128:35
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
note: run with `MADSIM_TEST_SEED=16` environment variable to reproduce this error
```

From what I see here, it seems like distributed scheduler fails, and has the correct error. But the query is still succeeding.

### To Reproduce

Not sure how to reproduce locally.

### Expected behavior

- For RW, it should return out of range error.
- Additionally, if task failed to schedule, it should never succeed.
- For our e2e test runner, we can include the success response in the debug statement.

### Additional context

Encountered here: https://buildkite.com/risingwavelabs/pull-request/builds/17838#01866f3e-d87f-4d3c-9f04-c2367ca93ff6

Logs: https://bulidkite-artifacts-bucket.s3.amazonaws.com/9eed51d0-eaaf-4ea3-95a2-2e9e557607f6/01866f3d-c3a2-4f1c-96e3-a5df76e1c094/01866f3e-d87f-4d3c-9f04-c2367ca93ff6/risedev-logs/batch-16.log

### PR DESCRIPTION
**Pull request by liurenjie1024**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As title.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

Fix #8066

### CODE REVIEW

File: e2e_test/batch/basic/query.slt.part [hunk 1]
```diff
@@ -30,6 +30,10 @@ select count(*) from t3;
 ----
 1
 
+statement error
```
Comment (lmatz) [inline e2e_test/batch/basic/query.slt.part:33]: Sqllogictest is able to test whether it matches the entire error message

**Review by lmatz — COMMENTED** [review]
In the distributed mode, I suppose that multiple upstreams can fail and thus output multiple error messages. 🤔 
We will choose the first one reported and return it to the user, right?

**Review by chenzl25 — COMMENTED** [review]
Could you please remove this line from `e2e_test/batch/issue_7324.slt`?
https://github.com/risingwavelabs/risingwave/blob/abe2ba32cfa84b65717871262318bd127a2c5c3d/e2e_test/batch/issue_7324.slt#L7

**Review by chenzl25 — APPROVED** [review]
LGTM!

**Pull request merged**